moleculemaker/mmli-backend
# fix: point local dev at staging, use correct values for prod/staging

## Problem
Some very minor issues need to be cleaned up in values files

## Approach
* Removed some stale TODO comments from values files
* Pin to latest and/or main Docker image, instead of random PR tag
* Map local source into container for Docker compose case

## How to Test
Currently deployed in [production](https://mmli.fastapi.mmli1.ncsa.illinois.edu/docs)

1. Navigate to https://mmli.fastapi.mmli1.ncsa.illinois.edu/docs
2. Scroll to the `/upload` endpoint and upload a PDF file containing molecular diagrams
    * You should see the upload completes successfully
3. Scroll to the `/analyze` endpoint and submit the same jobId that was returned from upload
4. Wait for the job to finish
    * You should see that the job completes successfully
5. Get the production username/password from the Kubernetes secret: `kubectl get secret -n alphasynthesis mmli-backend-minio -o yaml`
    * You should see `root-user` and `root-password` are printed here
6. Pass each of these into `base64 -d` to decode the username and password, e.g. `echo -n 'OGF...NmI=' | base64 -d`
7. Using these decoded credentials, Login to [production minio](https://minio.mmli.fastapi.mmli1.ncsa.illinois.edu/login)
8. Click down into the `chemscraper` bucket and find your `jobId`
    * You should see that your input/output files are stored here
